cantabular/dimensions.go

```diff
@@ -26,10 +26,45 @@ func (c *Client) GetDimensions(ctx context.Context, dataset string) (*GetDimensi
 		Dataset: dataset,
 	}
 
+	if err := c.queryUnmarshal(ctx, QueryAllDimensions, data, resp); err != nil {
+		return nil, err
+	}
+
+	if resp != nil && len(resp.Errors) != 0 {
+		return nil, dperrors.New(
+			errors.New("error(s) returned by graphQL query"),
+			resp.Errors[0].StatusCode(),
+			log.Data{"errors": resp.Errors},
+		)
+	}
+
+	return &resp.Data, nil
+}
+
+// GetDimensions performs a graphQL query to obtain all the non-geography dimensions for the provided
+// cantabular dataset. The whole response is loaded to memory.
```

Comment:
I suppose this is the only way? 
Are we aware of how big the responses can be?

Reply:
The size of the responses for lists of dimensions won't be significant enough to worry about memory, that's only really relevant for options and counts

Reply:
That comment is just a slightly modified version of what was there before